### PR TITLE
LTX Video training support

### PR DESCRIPTION
**Pull request by bghira**
- [x] a `--model_family=ltxvideo` argment is added
- [x] configure script is updated to handle new model
- [x] initial model loading
- [x] text embed caching
- [x] VAE caching
  - [x] images
  - [x] video (if applicable)
- [x] video aspect bucketing (`crop=False`)
- [x] checkpoint save/load
  - [x] full model
  - [x] LoRA
  - [x] final full model saving at end of training
  - [x] final lora model saving at end of training
- [x] any kind of specific loss augmentation - scaling, weighting, masking, must be added to the training loop if required
- [x] validations must load the pipeline classes (`helpers/training/validation.py`)
- [x] model card should be updated to include model-specific information
- [x] model card inference example should work correctly
- [x] data backends are updated to handle any new data types
  - [x] local data
  - [x] s3 data backend
  - [x] URL-based (csv) data loading
- [x] metadata backends are updated to handle any new required fields or file type logic
  - [x] discovery backend
  - [x] parquet backend
- [x] documentation must be produced and included, eg. a quickstart guide w/ a functional dataset is required for all new models added
- [x] block access to controlnet training
- [ ] block access to LoRA training
- [x] block access to text encoder training
- [ ] block access to `--validation_using_datasets` if it does not have img2img pipeline
  - [ ] implement image-to-image/image-to-video sampling if applicable
- [x] LoRA training
- [x] LyCORIS training

## nice-to-have, but not needed

- [ ] controlnet training / inference validations
- [ ] refiner mode for validations, eg. `denoising_start` support in pipeline

## final pre-merge testing

- [x] loading initial model
- [x] vae caching
  - [x] debug output saving during VAE transforms / pre-processing samples
- [x] text embed caching
- [x] saving a checkpoint
- [x] loading a checkpoint
- [x] final pipeline export
- [x] text2img validations
- [ ] img2img validations
- [ ] if implemented, refiner-based partial denoising validations (uses PIL inputs, sorry)
- [ ] min-snr-gamma compatibility
- [ ] offset noise compatibility
- [x] model card details
- [ ] model card inference code
- [ ] noise scheduler changes at validation time via `--validation_noise_scheduler`

### CODE REVIEW

**Comment by bghira** [comment]
it can now correctly crop videos and cache those vae entries.

batched video training works fine.

`crop=False` has issues with squished encoded videos, and non-square `crop=True` configurations aren't yet tested.

**Comment by bghira** [comment]
also adjusted the i2v logic to match the newer stuff from finetrainers which uses a probabilistic approach by default instead of full first-frame protection, as the earlier finetrainers version had done.

so we're at 10% I2V training by default w/ LTX, and 125 frames (5 seconds) of video, shorter videos are excluded and longer videos are just truncated.

`num_frames` is dataset-specific property. `i2v_data` can be set on a dataset to False to purposely exclude it from the usual i2v calculations if you prefer it to be t2v trained instead.

investigated the sampling of timesteps needed by the model and it's all good for now so just the aspect bucketing and video resize stuff remains to be checked out before it can be cleaned up and finalised.

**Comment by bghira** [comment]
i was able to catch the issue with the video aspect bucketing, at one point the current_size got updated to the `h, w` layout instead of `w, h` as images do, and resolving that so the values were correctly ordered made the downsample, crop and resize methods begin working correctly.

there were some other minor issues fixed but so far it's probably ready for prime-time testing by broader audience.

my config.json:

```json
{
    "caption_dropout_probability": 0.1,

    "data_backend_config": "config/ltxvideo/multidatabackend.json",

    "gradient_checkpointing": true,
    "lycoris_config": "config/ltxvideo/lycoris_config.json",

    "model_family": "ltxvideo",
    "model_type": "lora",
    "base_model_precision": "no_change",
    "lora_rank": 64,
    "lora_type": "lycoris",

    "optimizer": "optimi-lion",
    "learning_rate": "8e-5",

    "pretrained_model_name_or_path": "Lightricks/LTX-Video",

    "resolution": 480,
    "resolution_type": "pixel_area",

    "train_batch_size": 4,
    "vae_batch_size": 1,

    "validation_guidance": 4.2,
    "validation_num_inference_steps": "25",
    "validation_prompt": "A photo-realistic cat jumping through a field of lavender flowers in a mountainous region.",
    "validation_resolution": "384x256"
}
```

note that these aren't necessarily ideal values, especially the resolution, just that they're what i've been using to test.

```json
[
    {
        "id": "image-dataset-384",
        "type": "local",
        "instance_data_dir": "/Volumes/ml/datasets/test_datasets/single_image_dataset",
        "crop": false,
        "crop_style": "random",
        "minimum_image_size": 128,
        "resolution": 384,
        "resolution_type": "pixel_area",
        "repeats": 4,
        "metadata_backend": "discovery",
        "caption_strategy": "filename",
        "cache_dir_vae": "cache/ltxvideo-vae-384"
    },
    {
        "id": "video-dataset-384",
        "type": "local",
        "dataset_type": "video",
        "instance_data_dir": "/Volumes/ml/datasets/test_datasets/sora_videos",
        "crop": false,
        "crop_style": "random",
        "minimum_image_size": 384,
        "resolution": 384,
        "resolution_type": "pixel_area",
        "repeats": 4,
        "metadata_backend": "discovery",
        "caption_strategy": "filename",
        "cache_dir_vae": "cache/ltxvideo-vae-384-video",
        "video": {
            "num_frames": 125
        }
    },
    {
        "id": "text-embed-cache",
        "dataset_type": "text_embeds",
        "default": true,
        "type": "local",
        "cache_dir": "cache/ltxvideo-text"
    }
]
```

the `video` subsection isn't needed, but `dataset_type=video` would be if you have mp4 files inside there.

in general it's not tested what happens when you combine videos and images into one dataset. so for now, either expect it to break (and report this issue please) or just don't do that :D 

lycoris config:

```json
{
    "algo": "lokr",
    "multiplier": 1.0,
    "linear_dim": 10000,
    "bypass_mode": true,
    "linear_alpha": 1,
    "factor": 16,
    "apply_preset": {
        "target_module": [
            "Attention", "FeedForward"
        ],
        "module_algo_map": {
            "Attention": {
                "factor": 16
            },
            "FeedForward": {
                "factor": 8
            }
        }
    }
}
```
